src/main/java/seedu/address/logic/commands/TagAddCommand.java

```diff
@@ -77,16 +87,66 @@ public boolean equals(Object other) {
         }
 
         TagAddCommand e = (TagAddCommand) other;
-        return tags.equals(e.tags);
+        return tagsToAdd.equals(e.tagsToAdd);
     }
 
     /**
      * Generates a command execution success message based on whether
-     * the remark is added to or removed from
+     * the tag is successfully added.
      * {@code personToEdit}.
      */
-    private String generateSuccessMessage(Person personToEdit) {
-        String message = !tags.isEmpty() ? MESSAGE_ADD_TAG_SUCCESS : MESSAGE_DELETE_TAG_SUCCESS;
-        return String.format(message, Messages.formatForTags(personToEdit), Messages.getName(personToEdit));
+    public String generateSuccessMessage(Person personToEdit, Person editedPerson) {
+        Set<Tag> tagsInBoth = new HashSet<>(personToEdit.getTags());
+        Set<Tag> tagsInNeither = new HashSet<>(tagsToAdd);
+        // if all tags in og person matches the tags to add, means all tags to be added
+        // are duplicates, so dont go inside the loop
+        if (!personToEdit.getTags().containsAll(tagsToAdd)) {
+            tagsInBoth.retainAll(tagsToAdd); // duplicates that we dont want to add
+            tagsInNeither.removeAll(tagsInBoth); // new tags minus the duplicates that we want to add
+            if (tagsInBoth.isEmpty()) {
+                // if there are no duplicates, this is a clean addition
+                return String.format(MESSAGE_ADD_TAG_SUCCESS, Messages.tagSetToString(tagsToAdd),
+                        Messages.getName(editedPerson));
+            } else { // if there are some duplicates
+                // gets the tags that we actually want to add
+                String nonDuplicateTagsExist = String.format(MESSAGE_ADD_TAG_SUCCESS + "\n",
+                        Messages.tagSetToString(tagsInNeither), Messages.getName(editedPerson));
+                // gets the duplicate tags that we dont want to add
+                String duplicateTagsExist = String.format(MESSAGE_DUPLICATE_TAGS,
+                        Messages.getName(editedPerson), Messages.tagSetToString(tagsInBoth));
+                return nonDuplicateTagsExist + duplicateTagsExist;
+            }
+        }
+        return String.format(MESSAGE_DUPLICATE_TAGS, Messages.getName(editedPerson),
+                Messages.tagSetToString(tagsToAdd));
+    }
+
+    /**
+     * Handles duplicate tags as inputted by the user.
+     * Checks if any of the tags from user input matches existing tags in the
+     * original Person.
+     * Matching tags are added to a separate set that keeps track of duplicate tags.
+     * Unique tags are added to a separate list of tags.
+     * @param person EditedPerson from the execute method above.
```

Comment:
Leave a space between descriptor and parameter section (as per Java convention)

Reply:
Noted RSM. It wont happen again.